lib/zone.ts

```diff
@@ -1354,6 +1362,12 @@ const Zone: ZoneType = (function(global: any) {
       }
       error.stack = error.zoneAwareStack = frames.join('\n');
     }
+    // if this is valid, means not undefined
+    // and not global (non-strict mode call Error())
+    // and it is an error not an unrelated object
+    if (this && this !== global && this instanceof Error) {
+      error = setPrototypeOf(error, getPrototypeOf(this));
```

Comment:
Thank you for pointing out this one, I didn't realize that setPrototypeOf will make javascript engine optimization become slow.  how about change the code above to 

```javascript
    // if this is valid, means not undefined
    // and not global (non-strict mode call Error())
    // and it is an error not an unrelated object
    if (this && this !== global && this instanceof Error) {
       Object.assign(error, this);
       return this;
    }
```

-Pro: don't use setPrototypeOf
-Con:if/else will return different object